x/crosschain/keeper/abci.go

```diff
@@ -175,3 +204,81 @@ func CheckAndUpdateCctxGasPrice(
+	cctx types.CrossChainTx,
+	flags observertypes.GasPriceIncreaseFlags,
+) (gasPriceIncrease math.Uint, additionalFees math.Uint, err error) {
+	// check retry interval -- use default interval if not set
+	retryIntervalBTC := flags.RetryIntervalBTC
+	if retryIntervalBTC <= 0 {
+		retryIntervalBTC = observertypes.DefaultGasPriceIncreaseFlags.RetryIntervalBTC
+	}
+
+	// skip if Bitcoin gas price retry interval is not reached
+	lastUpdated := time.Unix(cctx.CctxStatus.LastUpdateTimestamp, 0)
+	if ctx.BlockTime().Before(lastUpdated.Add(retryIntervalBTC)) {
+		return math.ZeroUint(), math.ZeroUint(), nil
+	}
+
+	// get current gas price
+	currentGasPrice, err := cctx.GetCurrentOutboundParam().GetGasPriceUInt64()
+	if err != nil {
+		return math.ZeroUint(), math.ZeroUint(), err
+	}
+
+	// use latest median gas price as new gas price, the reasons are:
+	// 1. the goal is to increase the average gas price of all the stuck txs to market level
+	// 2. zetaclient can't replace stuck tx individually, it only gives more funds to the last stuck tx (child tx)
+	// 3. updating all pending CCTXs to the same 'mediaGasPrice' number simplyfies the calculation in zetaclient
+	newGasPrice := medianGasPrice
+	if math.NewUint(currentGasPrice).GTE(newGasPrice) {
+		return math.ZeroUint(), math.ZeroUint(), nil
+	}
+
+	// compute gas price increase
+	gasPriceIncrease = newGasPrice.SubUint64(currentGasPrice)
+
+	// withdraw additional fees from the gas stability pool
+	chainID := cctx.GetCurrentOutboundParam().ReceiverChainId
+	gasLimit := math.NewUint(cctx.GetCurrentOutboundParam().CallOptions.GasLimit)
+	additionalFees = gasLimit.Mul(gasPriceIncrease)
+	if err := k.fungibleKeeper.WithdrawFromGasStabilityPool(ctx, chainID, additionalFees.BigInt()); err != nil {
+		return math.ZeroUint(), math.ZeroUint(), cosmoserrors.Wrap(
+			types.ErrNotEnoughFunds,
+			fmt.Sprintf(
+				"cannot withdraw %s satoshis from gas stability pool, error: %s",
+				additionalFees.String(),
+				err.Error(),
+			),
+		)
+	}
+
+	// set new gas price and last update timestamp
+	//
+	// there is no priority fee for Bitcoin chain, the reason for setting 'GasPriorityFee' is:
+	// zetaclient can't figure out whether the gas price has been updated or not, so setting the
+	// 'GasPriorityFee' is to signal the gas price is updated, and the RBF (replace by fee) is allowed.
+	cctx.GetCurrentOutboundParam().GasPrice = newGasPrice.String()
+	cctx.GetCurrentOutboundParam().GasPriorityFee = newGasPrice.String()
+	k.SetCrossChainTx(ctx, cctx)
+
+	return gasPriceIncrease, additionalFees, nil
+}
+
+// IsCCTXGasPriceUpdateSupported checks if the given chain supports gas price update
+func IsCCTXGasPriceUpdateSupported(chainID int64, additionalChains []chains.Chain) bool {
+	switch {
+	case chains.IsZetaChain(chainID, additionalChains):
+		return false
+	case chains.IsEVMChain(chainID, additionalChains),
+		chains.IsBitcoinChain(chainID, additionalChains):
+		return true
+	default:
+		return false
```

Comment:
```suggestion
func IsCCTXGasPriceUpdateSupported(chainID int64, additionalChains []chains.Chain) bool {
	switch {
	case chains.IsEVMChain(chainID, additionalChains),
		chains.IsBitcoinChain(chainID, additionalChains):
		return true
	default:
		return false
```

Reply:
The `chains.IsEVMChain` also returns `true` on ZetaChain as ZetaChain has `chain.Vm == Vm_evm`.